### DISCUSSION TITLE
The future of Elemental-Toolkit & Kairos

### DISCUSSION THREAD

**Discussion by fullmetal-fred** [kairos-io/kairos#429] [General]
Hey @mudler, I'd like to better understand the use of elemental-toolkit behind Kairos. If I'm trying to make my own meta-distributions (that may or may not have anything to do with K8s), should I be starting with elemental-toolkit directly? Or Kairos? Can you characterize the community strength / trajectory at all? Are you forking and forgetting elemental-toolkit or planning on trying to merge with the original? Just trying to get a feel for what community I should be investing my time into. 

Thanks!

**Comment by mudler** [comment]
Hi @cybernetic-mountaineer !

We are using a fork of Elemental-toolkit at version https://github.com/rancher/elemental-toolkit/releases/tag/v0.8.14-22. As much as I dislike forking, this happened for technical reasons:

- Elemental's goals and requirements seems to switched dramatically and impacting Kairos by looking at the latest changes. Indeed there is no BIOS/EFI hybrid support anymore in new versions - and there was no way to interact with the development team before such breaking changes hit the project. This from a Kairos perspective is problematic, as not having an open channel and be able to influence Elemental-toolkit roadmap have a big impact on our development plans.
- Seems the distro-agnostic approach got much less attention, focusing more on openSUSE/SUSE distributions - while this is completely understandable and fair - last time I've tried to consume Elemental new releases where only working on openSUSE/SUSE.
- The Elemental-toolkit project seems to be more in maintainance mode to my eyes - this is a personal take from looking at elemental-toolkit commits history, and tags/changes between releases and the team board https://github.com/orgs/rancher/projects/9/views/1 that doesn't have any toolkit items almost anymore, which goes a bit off-track with Kairos plans to customize images.

I've left SUSE where I was leading the Elemental team so to be honest I can't answer which project works better for your case - I can just talk about Kairos plans in #116 to streamline derivatives creation, and from a Kairos perspective being distro-agnostic is a key factor as we are not tied to any specific distribution. It is quite the opposite - we want to be able to convert any Distribution to our model.

I don't see in the future us merging back as there are many clashing points already between the twos, especially regarding distro compatibilities. I see that we will have our own CLI, and a streamlined strategy to build derivatives - however we strive to keep compatibility with Elemental and actively engage with them and help as we can.

**Reply by fullmetal-fred** [reply]
Wow, thanks @mudler for the clarification and thoughts. It all makes sense to me and I have to admit that I've been far more impressed with Kairos already. The distro-agnostic approach is really important to me as well, so I would say our goals align there. I'll stay tuned...excited to see the future of Kairos! Good luck 🚀
